src/Workspaces/CSharp/Portable/Formatting/Rules/BaseFormattingRule.cs

```diff
@@ -186,83 +186,10 @@ protected void AddBraceSuppressOperations(List<SuppressOperation> list, SyntaxNo
-            // lines.  The difference here is that because the user just hit close-curly here we can 
-            // consider it as being part of the closest construct and we can consider its placement
-            // when deciding if the construct is on a single line.
-
-            var endToken = bracePair.Item2;
-            if (lastToken.Kind() != SyntaxKind.CloseBraceToken &&
-                lastToken.Kind() != SyntaxKind.EndOfFileToken &&
-                !endToken.IsMissing)
-            {
-                // The user didn't just type the close brace.  So any close brace we have may 
-                // actually belong to a containing construct.  See if any containers are missing
-                // a close brace, and if so, act as if our own close brace is missing.
-
-                if (SomeParentHasMissingCloseBrace(node.Parent))
-                {
-                    if (node.IsKind(SyntaxKind.Block) && ((BlockSyntax)node).Statements.Count >= 1)
-                    {
-                        // In the case of a block, see if the first statement is on the same line 
-                        // as the open curly.  If so then we'll want to consider the end of the
-                        // block as the end of the first statement.  i.e. if you have:
-                        //
-                        //  try { }
-                        //  catch { return;     // <-- the end of this block is the end of the return statement.
-                        //  Method();
-                        var firstStatement = ((BlockSyntax)node).Statements[0];
-                        if (FormattingRangeHelper.AreTwoTokensOnSameLine(firstTokenOfNode, firstStatement.GetFirstToken()))
-                        {
-                            endToken = firstStatement.GetLastToken();
-                        }
-                    }
-                    else
-                    {
-                        endToken = endToken.GetPreviousToken();
-                    }
-                }
-            }
-
             // suppress wrapping on whole construct that owns braces and also brace pair itself if 
             // it is on same line
-            AddSuppressWrappingIfOnSingleLineOperation(list, firstTokenOfNode, endToken);
-            AddSuppressWrappingIfOnSingleLineOperation(list, bracePair.Item1, endToken);
-        }
-
-        private bool SomeParentHasMissingCloseBrace(SyntaxNode node)
-        {
-            while (node != null && node.Kind() != SyntaxKind.CompilationUnit)
-            {
-                var bracePair = node.GetBracePair();
-                if (bracePair.Item2.IsMissing)
-                {
-                    return true;
-                }
-
-                node = node.Parent;
-            }
-
-            return false;
+            AddSuppressWrappingIfOnSingleLineOperation(list, firstTokenOfNode, bracePair.Item2);
+            AddSuppressWrappingIfOnSingleLineOperation(list, bracePair.Item1, bracePair.Item2);
```

Comment:
:bulb: Implementation is outside the scope of this pull request, but we should file an issue to convert existing uses of `ValueTuple<T1, T2>` and other generic tuples to the new tuple syntax with named tuple elements. We should also make sure an issue exists to create an analyzer to aid in the conversion.